src/components/BivariateLegend/BivariateLegend.tsx

```diff
@@ -115,3 +114,29 @@ export function BivariateLegend({
     </div>
   );
 }
+
+const formatBivariateAxisLabel = (quotients: Axis['quotients']): string => {
+  if (!quotients) return '';
+  const [nominator, denominator] = quotients;
+
+  // we don't have nominator unit - don't show units
+  if (isNilOrOther(nominator.unit.id)) {
+    return `${nominator.label} to ${denominator.label}`;
+  }
+  // we have nominator unit and denominator unit is - one
+  if (denominator.name === 'one') {
+    return `${nominator.label} (${nominator.unit.shortName})`;
+  }
+  // we have nominator unit, but don't have denominator unit - show only nominator unit
+  if (isNilOrOther(denominator.unit.id)) {
+    return `${nominator.label} to ${denominator.label} (${nominator.unit.shortName})`;
+  }
+  // here we process cases where we have both units
+  if (denominator.name === 'area_km2') {
+    return `${nominator.label} (${nominator.unit.shortName}/${denominator.unit.shortName})`;
+  }
```

Comment:
great thing that you've used comments (bivariate stuff is hard by itself)
but you keep repeating "we have\don't have nominator unit"
maybe there's a way to tell that via code naming

Reply:
I have no idea how. If you have idea - please suggest.
Every 'if' just cuts down cases before 'if', that's why I've decided to describe what's covered in every case for easier navigation and understanding.

Reply:
i've made them shorter, hope it will be ok